drivers/sensor/sensor_sim/sensor_sim.c

```diff
@@ -203,21 +293,77 @@ static double generate_pseudo_random(void)
 }
 
 /*
- * @brief Calculates sine from uptime
- * The input to the sin() function is limited to avoid overflow issues
+ * @brief Generate value for acceleration signal toggling between two values on fetch.
  *
- * @param offset Offset for the sine.
- * @param amplitude Amplitude of sine.
+ * @param chan[in]	Selected sensor channel.
+ * @param out_val[out]	Pointer to the variable that is used to store result.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
  */
-static double generate_sine(double offset, double amplitude)
+static int generate_toggle(enum sensor_channel chan, double *out_val)
 {
-	/* Predefined period for generated sine function. */
-	static const uint32_t period_ms = 10000;
+	static const double amplitude = 20.0;
+	static double val_sign = 1.0;
+	double res_val = amplitude * val_sign;
+
+	*out_val = res_val;
+
+	if (chan == SENSOR_CHAN_ACCEL_XYZ) {
+		*(out_val + 1) = res_val;
```

Comment:
The generator_functions use a common API and number of doubles is defined by the selected channel (in the same way as in Zephyr's sensor API).
Maybe I could add small inline function that converts the channel to number of required values instead? I could use it also for setting the wave parameters, so it would be consistent too.

Reply:
Discussced offline - added size as an argument